.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,61 @@
+name: Integration tests
+on:
+  pull_request:
+  push:
+    branches: [main]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        api-level: [33] # TODO: add more api levels to test
+    env:
+      MAILOSAUR_API_KEY: ${{ secrets.MAILOSAUR_API_KEY }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: Setup JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
```

Comment:
This was the minimum JDK version with which the emulator worked.

We can leave it as-is, or test against various versions using the matrix pattern, if that's useful.